lms/djangoapps/discussion_api/api.py

```diff
@@ -751,20 +751,28 @@ def _do_extra_actions(api_content, cc_content, request_fields, actions_form, con
                 raise ValidationError({field: ["Invalid Key"]})
 
 
-def _handle_following_field(form_value, user, cc_content):
+def _handle_following_field(form_value, user, cc_content, api_content):
     """follow/unfollow thread for the user"""
     if form_value:
         user.follow(cc_content)
+        if cc_content["type"] == "thread":
+            api_content["read"] = True
```

Comment:
It feels like it could easily be missed.  
What about the following?:
```
# always treat the thread as read as a reasonably accurate shortcut, rather than adding additional processing.
treat_as_read = True
...
api_content["read"] = treat_as_read
```